src/model_editor/AccessPolicyStore.cpp

```diff
@@ -131,26 +128,28 @@ bool AccessParser::startElement(const QString& /*namespaceURI*/, const QString&
       }
     }
     return false;  //NO IddObjectType!!!!
-  } else if (!qName.compare("rule", Qt::CaseInsensitive)) {
-    if (!m_curPolicy) {
+  }
+
+  if (qName.compare("rule", Qt::CaseInsensitive) == 0) {
+    if (m_curPolicy == nullptr) {
       LOG(Debug, "parse error, rule started before a policy is started");
       return false;
     }
     QString fieldName;
     QString accessRule;
     for (int i = 0, iend = atts.length(); i < iend; ++i) {
       QString name = atts.qName(i);
-      if (!name.compare("IddField", Qt::CaseInsensitive)) {
+      if (name.compare("IddField", Qt::CaseInsensitive) == 0) {
         fieldName = atts.value(i);
-      } else if (!name.compare("access", Qt::CaseInsensitive)) {
+      } else if (name.compare("access", Qt::CaseInsensitive) == 0) {
         accessRule = atts.value(i);
       }
     }
-    if (fieldName.size() && accessRule.size()) {
+    if (!fieldName.isEmpty() && !accessRule.isEmpty()) {
       AccessPolicy::ACCESS_LEVEL level = AccessPolicy::FREE;
-      if (!accessRule.compare("locked", Qt::CaseInsensitive)) {
+      if (accessRule.compare("locked", Qt::CaseInsensitive) == 0) {
         level = AccessPolicy::LOCKED;
-      } else if (!accessRule.compare("hidden", Qt::CaseInsensitive)) {
+      } else if (accessRule.compare("hidden", Qt::CaseInsensitive) == 0) {
```

Comment:
Did clang tidy make this change or did you have to?

Reply:
It did for me. QString::compare returns an int.

(Also, aside from int -> bool conversion, `!str.compare("other")` to express "if it matches" is HELLA confusing to me)

Reply:
Seems like a good change